syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -470,3 +472,113 @@ def clone_from_state(self, state: dict):
     @property
     def debug_log(self):
         return self._debug_log
+
+
+class GridSearcher(BaseSearcher):
+    def __init__(self, config_space, metric, points_to_evaluate=None, **kwargs):
+        super().__init__(config_space, metric, points_to_evaluate)
+        self._hp_ranges = make_hyperparameter_ranges(config_space)
+        self._excl_list = ExclusionList.empty_list(self._hp_ranges)
+        self._config_candidates = self._generate_config_candidates(config_space)
```

Comment:
This could potentially become pretty big. It is probably OK, but a nicer solution would create the grid entries on demand.

Reply:
Nit: `_config_candidates` is a bit generic and does not explain what those are. Per our discussion - could be `_grid_candidates` or `_all_candidates_on_grid` or something along the lines of this.

Reply:
Rename the function as `_generate_all_candidates_on_grid()`, and class member to `_remain_candidates_on_grid`.
Since the scheduler will pop items from the list whenever `get_config()` is called, I name it begin with `_remain`